s4e-backend/src/main/java/pl/cyfronet/s4e/sync/step/VerifyAllArtifactsExist.java

```diff
@@ -0,0 +1,67 @@
+package pl.cyfronet.s4e.sync.step;
+
+import lombok.Builder;
+import lombok.val;
+import pl.cyfronet.s4e.ex.S3ClientException;
+import pl.cyfronet.s4e.service.SceneStorage;
+import pl.cyfronet.s4e.sync.Error;
+import pl.cyfronet.s4e.sync.context.Context;
+
+import javax.json.JsonObject;
+import javax.json.JsonString;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static pl.cyfronet.s4e.sync.Error.*;
+
+@Builder
+public class VerifyAllArtifactsExist implements Step<Context, Error> {
+    public static final String SCENE_ARTIFACTS_PROPERTY = "artifacts";
+
+    private final Supplier<SceneStorage> sceneStorage;
+
+    private final Function<Context, JsonObject> sceneJson;
+    private final BiConsumer<Context, Map<String, String>> update;
+
+    @Override
+    public Error apply(Context context) {
+        val error = context.getError();
```

Comment:
Those 3 lines of code I see 3rd time, maybe it can be moved to some method?

val error = context.getError();
SceneStorage sceneStorage = this.sceneStorage.get();
JsonObject sceneJson = this.sceneJson.apply(context);

Reply:
I know it is repetible, but as each Step has the input passed in form of lambdas to ensure thread-safety it is necessary to do there. Also, such structuring of the local variables assignments makes it easier to follow what are each step's dependencies, input and output.

Reply:
That's fine